### ISSUE TITLE
Error on Polling

### ISSUE THREAD

**Issue by abillauer** [ajslater/codex#445]
I have added some new books to the library and I am seeing the error below for each book. I do not see the book added afterwards. (codex 1.7.5 on docker)

```

2024-11-09 22:02:38 UTC ERROR   Error preparing /comics/xxx.cbz for create.
Traceback (most recent call last):
  File "/usr/local/lib/python3.13/site-packages/codex/librarian/importer/create_comics.py", line 111, in bulk_create_comics
    comic.presave()
    ~~~~~~~~~~~~~^^
  File "/usr/local/lib/python3.13/site-packages/codex/models/comic.py", line 223, in presave
    super().presave()
    ~~~~~~~~~~~~~~~^^
  File "/usr/local/lib/python3.13/site-packages/codex/models/groups.py", line 137, in presave
    WatchedPath.presave(self)
    ~~~~~~~~~~~~~~~~~~~^^^^^^
  File "/usr/local/lib/python3.13/site-packages/codex/models/paths.py", line 49, in presave
    self.set_stat()
    ~~~~~~~~~~~~~^^
  File "/usr/local/lib/python3.13/site-packages/codex/models/paths.py", line 32, in set_stat
    st_record = path.stat()
  File "/usr/local/lib/python3.13/pathlib/_local.py", line 515, in stat
    return os.stat(self, follow_symlinks=follow_symlinks)
           ~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/comics/xxx.cbz'

```
The files are present in the system and have the same owner/permissions as the other files. Am I doing something wrong or is this a bug in the system?

As an aside the polling has been much faster in the latest releases! Thanks and keep up the great work!

**Comment by ajslater** [comment]
This is very strange.
- Import reads the comic archive successfully and collects tags from it without error and passes the comic path on to the next import steps.
- Import gets all the way to create_comic which happens after create tags.
- The create_comic step then reads the filesystem stat info to record the comic mtime but that reports that the comic does not exist.

Possibilities:
- There is a bug where the path of the file is changed in import and the that path truly does not exist. But If I read you correctly you can verify from the error message that the path is indeed correct.
- The filesystem mount is somehow unstable and flickers out when import goes to read the stat. In that case I'd expect the error to be intermittent, but that doesn't seem to be the case.

I am guessing that blame rests somewhere within the filesystem mount, probably with docker or docker interacting with a network or unusual filesystem, but I really don't know. Honestly I'm baffled. I'd try restarting all of docker. And if network filesystems are involved, remounting those. I'll think on this, but I have no immediate answers.

**Comment by abillauer** [comment]
I downed the docker, rebooted, the machine, and nothing helped. 

I can confirm I can see and read other comics. Even ones in the same directory. The mount is remote over NFS. I can confirm that the filesystem is stable. I can cat out the comic files that cause the error on the command line of the host machine. The only books that seem impacted are the from the last batch that were added to the server

I just added other new files and they are ingested without error. The error persists on the other books that were added last time.

I removed the files that are causing the error and re-added one of them and still got the same error. Other new files seem to be fine. It seems it may be an issue with the specific files. I will continue to investigate.

**Comment by abillauer** [comment]
After further testing I think the error has to do with having an "&" character in the file name. Older comics that were imported with a "&" the name are fine. New comics with an "&" are getting the error.

I tried importing a comic with the "&" character and it failed. I renamed it to the word "and" and it worked no problem.

**Comment by abillauer** [comment]
Confirmed. I renamed all the files that failed (all had "&" characters) and the poll/import worked as expected. If I rename them to have an "&" they fail with the error above.

**Comment by ajslater** [comment]
I think I can replicate! Thank you!

**Comment by ajslater** [comment]
Fixed in[ v1.7.6](https://github.com/ajslater/codex/releases/tag/v1.7.6).
I went hog wild with some string security sanitizing in 1.7.5 and I had the file path on the list. This also probably wrecked codex for utf-8 filenames.

**Comment by abillauer** [comment]
Confirmed! Thanks.
